File: java/src/Problem1.java
/**
 * Created by adam on 2/1/14.
 */
public class Problem1 implements Euler{
    static final int limit = 1000;
    static final int minimum = 3;
    /** The answer to problem 1
     *
     * @return the answer
     */
    public String answer() {
        int sum = 0;
        for (int i = minimum; i < limit; i++) {
           if (i % 3 == 0 || i % 5 == 0) {
              sum += i;
           }
        }
        return Integer.toString(sum);
    }
}

File: dart/lib/src/Problem1.dart
import 'Euler.dart';

class Problem1 implements Euler {
  String answer() {
    String answer_h(int upto) {
      var list = new List<int>.generate(upto - 1, (i) => i + 1);
      var sum = 0;
      for (int i in list) {
        if ((i % 3 == 0) || (i % 5 == 0)) {
          sum += i;
        }
      }
      return sum.toString();
    }

    return answer_h(1000);
  }
}
